### PR TITLE
Add self evaluation file

### PR DESCRIPTION
**Pull request by adumortier**
Added the self evaluation README.md file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+## **Ruby Conventions and Mechanics**
+[x] Code is mostly properly indented, spaced, and lines are not excessively long.        
+[x] Class, method, variable, and file names follow convention.   
+[x] At least one hash is implemented in a way that makes logical sense.  <br>    \
+*Hashes are used to return the result from the encrypt and decrypt methods. The choice of a hash is justified because the order in which different parts of the output are returned does not matter.* .  <br>    \
+[x] Classes, methods, and variables are well named so that they clearly communicate their purpose.  \
+[x] Code is all properly indented and syntax is consistent.  \
+[x] No methods are longer than 10 lines long.   <br>    \
+*the longest method is 10 lines long (find_key_per_date in the Cracking class)* .  <br>    \
+[x] Most enumerables/data structures chosen are the most efficient tool for a given job,
```
Comment (adumortier) [inline README.md:26]: will need to be demonstrated during project eval

File: README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## **Functionality** 
+[x] `Enigma` Class with `encrypt` and `decrypt` methods successfully implemented. \
+[x] `Encrypt`/`decrypt` command line interfaces successfully implemented \
+[x] `Cracking` method and command line interface successfully implemented
+_____________
+## **Object Oriented Programming**
+[x] Project is broken into logical components that are appropriately encapsulated.   
+[x] No classes are unreasonably small or large .  
+[x] Student can articulate the single responsibilities of the various components.  \
```
Comment (adumortier) [inline README.md:11]: will need to be demonstrated during project eval

File: README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+*Hashes are used to return the result from the encrypt and decrypt methods. The choice of a hash is justified because the order in which different parts of the output are returned does not matter.* .  <br>    \
+[x] Classes, methods, and variables are well named so that they clearly communicate their purpose.  \
+[x] Code is all properly indented and syntax is consistent.  \
+[x] No methods are longer than 10 lines long.   <br>    \
+*the longest method is 10 lines long (find_key_per_date in the Cracking class)* .  <br>    \
+[x] Most enumerables/data structures chosen are the most efficient tool for a given job,
+_____________
+## **Test Driven Development** 
+[x] Every method is tested at both the unit and integration level and completely verify expected behavior .   
+[x] Obvious edge cases are addressed. git history demonstrates students are writing tests before implementation code.   <br>    \
```
Comment (adumortier) [inline README.md:30]: will need to be demonstrated during project eval

File: README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+[x] Test coverage metrics show coverage at 100%  \
+[x] Mocks and/or stubs are used appropriately to ensure testing is more robust  <br>    \
+*see tests for `Offset` and `Key` class*   <br>    \
+[x] Mocks and/or stubs are used appropriately to ensure testing is more efficient <br>  
+*see tests for the `Cracking` class* <br>
+_____________
+## **Version Control**
+[x] At least 40 commits.  <br>    \
+*58 commits*  <br>    \
+[x] All pull requests include related and logical chunks of functionality   <br>    \
```
Comment (adumortier) [inline README.md:41]: will need to be demonstrated during project eval
